### ISSUE TITLE
Make Keyboard laser interaction the default option

### ISSUE THREAD

**Issue by alexderpyfox** [overte-org/overte#580]
H

**Comment by JulianGro** [comment]
Personally I am against this, because I have exactly the opposite opinion.
Laser pointers are really annoying, since you have to aim at the keyboard. Mallets are by far the best keyboard input method for VR I have ever seen. The reason most other applications use lasers, is that it is the easiest to implement.

**Comment by vegaslon** [comment]
I agree mallets are a far better input device to get accurate typing in VR. Using a 2d input method against a 3d keyboard is not optimal.

**Comment by alexderpyfox** [comment]
Let's hear what new users opinion is on it

**Comment by ksuprynowicz** [comment]
Personally I prefer mallets, they make typing in VR a lot easier for people like me with damaged nerves in hands.
I think that it would be best to do a poll on this. I will try to organize one.

**Comment by alexderpyfox** [comment]
The setting won't disappear ofc only the default is changed then so for people who prefer mallets they can still use that

**Comment by JulianGro** [comment]
If this is controversial, we could bring it to the assembly next month. Stuff like this is pretty much what the organization is for.
Personally I also don't think that specifically asking new users would be a good idea, because like mentioned above, it is more common to have lasers, and people tend to prefer what they know, regardless of how much better the other option might or might not be.

**Comment by alexderpyfox** [comment]
Exactly they would want lasers because it's the norm but having an option to use the proprietary mallets it's always good for the reason ksuprynowicz stated

**Comment by alexderpyfox** [comment]
It's not that hard for an experienced user to switch to mallets but a lot harder for new users to switch to laser option the goal is to make it easier for new users then later they can chose mallets in the settings if they figured out how to get there

**Comment by ksuprynowicz** [comment]
Coming from VRchat I didn't notice them being difficult - it was immediately obvious how to use them and I found them easier to use than VRchat keyboard. We will do a poll and let community decide.

**Comment by AnotherFoxGuy** [comment]
I think it would be better to add it as an option to choose on the setup screen in the tutorial world, so users can decide when they launch the interface for the first time

**Comment by ksuprynowicz** [comment]
> I think it would be better to add it as an option to choose on the setup screen in the tutorial world, so users can decide when they launch the interface for the first time

That would be the best way, along with asking for other preferences, like smooth/step turning for example.

**Comment by AleziaKurdis** [comment]
Any other things you would like to have in the tutorial wizard ?
- Use Virtual Keyboad (Laser / Mallets)
- Rotation Mode (Snap Turn / Smooth Turn)

**Comment by vegaslon** [comment]
Mic check

**Comment by AleziaKurdis** [comment]
I started to add this to the tutorial Wizard...
Now we have to explain what "laser" and "mallets are, since at this step of the onboarding process, there are lot of chances that the new user has not even used the tablet, and even less used the virtual keyboard. (which is for the VR usecase only.)

I can put images... but how much this will really help to figure what it really implies?

I'm now questioning if it is that much an issue to default on one or the other. 
People will have to do their setup at some point, and probably not deeper than that at their arrival. 

Both option (laser and mallets) are functional to start in my opinion, **after**, it's a question of taste that you can only figure after have used a bit the tools. This is not "essential" to start.

**I would prefer at this point that we vote for what the default should be. (rather than add this to the wizard)**


I was also to add : 
- Rotation Mode
- Dominant Hand

But frankly, this wizard need to stay short and simple, just the essential to go in-world. 
Why those settings more than others? You can start with any of these option without getting hurt. I guess this is not that much 
essential to start .


Regarding the **mic check**, It would be useful... 
I was initially planning to add a sound meter
and the button to open the audio setup. (In the case the sound doesn't output or it is muted)
(I don't think it would be a good idea to duplicate all that very complex setup in the wizard.)
But if people open the audio config, they might not complete the wizard after that.
I might instead add it **after the completion**. Ex: ...Basic setup is done, now test your mic...


(any thought?)

**Comment by Armored-Dragon** [comment]
Personally I think it would be a good idea generally to have lasers on by default. The issue I run into is that lasers are really good only for 2D interfaces that are otherwise common in VR UI. Overte currently has a 3D keyboard and using that keyboard is very annoying using a laser. Using a 3D object to interact with 3D UI feels a lot better to me than using lasers.
